### PR TITLE
PrimitiveBatchData for all

### LINKED ISSUE #456: Optimize the PrimitiveBatchData enum
This enum has a Vec<PrimitiveInstance> that is the same for many batch types. Previously, they were separate types. We could unify this enum to use a single Vec<PrimitiveInstance> and then store offsets / lengths for each batch. This would allow us to change the renderer to create far fewer UBOs which would improve the speed of the compositor time.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #456 

More stuff to come.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/609)

### CODE REVIEW

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -1505,13 +1528,10 @@ impl PrimitiveBatch {
                       info: MixBlendMode) -> bool {
         match &mut self.data {
             &mut PrimitiveBatchData::Composite(ref mut ubo_data) => {
-                ubo_data.push(PackedCompositePrimitive {
```
Comment (glennw) [inline webrender/src/tiling.rs]: Should be able to remove the PackedCompositePrimitive struct now.

File: webrender/src/tiling.rs [hunk 2]
```diff
@@ -1485,13 +1511,10 @@ impl PrimitiveBatch {
                     LowLevelFilterOp::Opacity(amount) => (8, amount.to_f32_px()),
                 };
 
-                ubo_data.push(PackedBlendPrimitive {
```
Comment (glennw) [inline webrender/src/tiling.rs]: Should be able to remove the PackedBlendPrimitive struct now.

**Comment by kvark** [comment]
All done.

**Pull request merged**